pkg/clients/s3/bucket.go

```diff
@@ -211,3 +212,42 @@ func UpdateBucketACL(ctx context.Context, client BucketClient, bucket *v1beta1.B
 	_, err := client.PutBucketAclRequest(config).Send(ctx)
 	return err
 }
+
+// CopyTag converts a local v1beta.Tag to an S3 Tag
+func CopyTag(tag *v1beta1.Tag) *s3.Tag {
```

Comment:
I'm guessing that you implemented this function to not have to have that `if tag == nil` check each time you need it. In such cases where the logic that's being abstracted is really simple, I usually lean towards doing it in place so that it's easier to read and know what it does immediately without changing the context and checking what that function does, hence, would prefer duplication over reuse.

Reply:
Sounds good, I have made this change for CopyTag and CopyTags

Reply:
Since we get that `if` out, I think we can get rid of this function and just write the following where it's called  like:
```
rule.Filter.Tag = &s3.Tag{
		Key:   aws.String(tag.Key),
		Value: aws.String(tag.Value),
	}
```